### DISCUSSION TITLE
Where would be 'type': 'blender' plugin (for shape)?

### DISCUSSION THREAD

**Discussion by sunilhadap** [mitsuba-renderer/mitsuba-blender#96] [Q&A] [answered]
Hi,

As I was going through the mitsuba-blender plugin code, I came across what looks like a blender plugin shape type ('type': 'blender') - mitsuba-blender>io>exporter>geometry.py:26. Where is this shape plugin defined? Is it built into Mitsuba? I don't find any mention of it in the Mitsuba documentation. Tried to search mitsuba-blender codebase, but could not find any register_mesh either!

Regards

Sunil

**Comment by njroussel** [comment, marked as answer]
Hi @sunilhadap 

It is defined in Mitsuba 3,  in this file: https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/shapes/blender.cpp.
It's not mentioned in the documentation because it is only meant for "internal" use. As the name suggests, it's something that is specifically built for the Blender addon and wasn't meant to be used for anything else. In addition, with the high-frequency updates of the Blender Mesh representations, that plugin is subject to a lot of breaking changes.
I guess my point is that if you want to rely on that code for something, there is a strong risk that overtime it will no longer fit your initial purpose.

**Reply by sunilhadap** [reply]
Roger that. Thanks for your quick reply.
